lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -1091,10 +1090,30 @@ namespace Js
             }
             else
             {
-                return object && object->HasOwnProperty(propertyId);
+#ifdef DEBUG
+                // In debug builds, calculate expectedAnswer up-front so that
+                // we can Assert the fastpath gets the right result.
+                BOOL expectedAnswer = object && object->HasOwnProperty(propertyId);
+#endif
+
+                PropertyString *propString = requestContext->GetPropertyString(propertyId);
+                const PropertyCache *propCache = propString->GetPropertyCache();
+
+                if (object->GetType() == propCache->type)
+                {
+                    // The type cached for the property was the same as the type of this object
+                    // (i.e. obj in obj.hasOwnProperty), so we know the answer is "true".
+                    Assert(expectedAnswer == TRUE); // sanity check on the fastpath result
+                    return TRUE;
+                }
+
+#ifndef DEBUG
+                // In release builds, wait until here to calculate the result.
+                BOOL expectedAnswer = object && object->HasOwnProperty(propertyId);
```

Comment:
>BOOL expectedAnswer =  [](start = 16, length = 22)

No need to BOOL expectedAnswer, just return object && object->HasOwnProperty(propertyId);